clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -9133,8 +9157,10 @@ void SYCLPostLink::ConstructJob(Compilation &C, const JobAction &JA,
 
   // Turn on Dead Parameter Elimination Optimization with early optimizations
   if (!(getToolChain().getTriple().isAMDGCN()) &&
+      !TCArgs.hasArg(options::OPT_fsycl_link_EQ) &&
```

Comment:
What is it that causes the incompatibility? Potential FPGA backend issues for archives have sprung to my mind, but _the truth_ is better than my vague hypothesis :)

Reply:
Thanks for calling this out - it may be the improper approach here.  I didn't fully realize the potential FPGA impact in doing the check this way.  We may just want to update the tests that are using `-fsycl-link` in separate compilation scenarios to disable DAE.